src/lib/WalletStore.ts

```diff
@@ -88,79 +54,60 @@ export class WalletStore {
     }
 
     public async get(id: string): Promise<WalletInfo | null> {
-        const db = await this.connect();
-        const transaction = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME]);
-        const request = transaction.objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME).get(id);
-        const result = await WalletStore._requestAsPromise(request, transaction);
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        const result = await super.get(id);
         return result ? WalletInfo.fromObject(result) : result;
     }
 
     public async put(walletInfo: WalletInfo) {
-        const db = await this.connect();
-        const transaction = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME], 'readwrite');
-        const request = transaction.objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME).put(walletInfo.toObject());
-        return WalletStore._requestAsPromise(request, transaction);
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        return super.put(walletInfo.toObject());
     }
 
     public async remove(id: string) {
-        const db = await this.connect();
-        const transaction = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME], 'readwrite');
-        const request = transaction.objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME).delete(id);
-        return WalletStore._requestAsPromise(request, transaction);
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        return super.remove(id);
     }
 
     public async list(): Promise<WalletInfoEntry[]> {
-        const db = await this.connect();
-        const request = db.transaction([WalletStore.DB_ACCOUNTS_STORE_NAME], 'readonly')
-            .objectStore(WalletStore.DB_ACCOUNTS_STORE_NAME)
-            .openCursor();
+        this._storeName = WalletStore.DB_ACCOUNTS_STORE_NAME;
+        return super.list()  as Promise<WalletInfoEntry[]>;
+    }
 
-        const result: WalletInfoEntry[] = await WalletStore._readAllFromCursor(request);
-        return result;
+    public getConstants(): StoreConstants {
+        return {
+            DB_VERSION: 1,
+            DB_NAME: 'nimiq-hub',
+            DB_STORE_NAME: this._storeName,
+        };
+    }
+
+    protected upgrade(request: any, event: IDBVersionChangeEvent): void {
+        const db = request.result;
+        if (event.oldVersion < 1) {
+            // Version 1 is the first version of the database.
+            db.createObjectStore(WalletStore.DB_ACCOUNTS_STORE_NAME, { keyPath: 'id' });
+            db.createObjectStore(WalletStore.DB_META_DATA_STORE_NAME, { keyPath: 'name' });
+        }
     }
 
-    public async close() {
-        if (!this._dbPromise) { return; }
-        // If failed to open database (i.e. dbPromise rejects) we don't need to close the db
-        const db = await this._dbPromise.catch(() => null);
-        this._dbPromise = null;
-        if (db) { db.close(); }
+    protected toEntry(walletInfoOrValue: WalletInfoEntry): WalletInfoEntry {
```

Comment:
This is not actually correctly typed as meta data also gets passed through this.
Also `walletInfoOrMetaData` would be a better name.
```suggestion
    protected toEntry<T>(walletInfoOrMetaData: T): T {
```

Reply:
renamed and typed to `any`.

Reply:
I typed it according to your suggestion as it is more precise (even though it does not make a difference in the end, I think).